editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -7561,6 +7561,7 @@ void Node3DEditor::_add_sun_to_scene(bool p_already_added_environment) {
 	}
 	ERR_FAIL_NULL(base);
 	Node *new_sun = preview_sun->duplicate();
+	new_sun->set_name(adjust_name_casing(new_sun->get_name()));
```

Comment:
Did you not mean to fix the same issue for the Add Environment to Scene button? (It's created on the method right below this one).

Reply:
I am also just a first contributor, but putting my insight since I also looked at the issue without being aware there was a PR up already (And thought about doing this same solution):

Would it be preferred to fix the name for the WorldEnvironment on the calling point instead of modifying the Node implementation directly? I see that it is critical code, and it may not be worth the larger impact for an Editor-only fix. Along with the potential perf considerations.

Reply:
My thoughts exactly. As [I commented above](https://github.com/godotengine/godot/pull/87659#discussion_r1468838556), while the above solution may be more encompassing it would also affect code outside of the editor (even at run-time), which may have some unintended consequences. In this case it may be better to be "safe" and implement the same solution as the Sun node.